docs/mp/guides/migration_3x.adoc

```diff
@@ -0,0 +1,172 @@
+
+
+* *JSON-P (Jakarta JSON Processing) {version-lib-jakarta-jsonp-api}*:
++
+Moved to `jakarta` namespace.
+
+* *Jakarta Annotations {version-lib-jakarta-annotations-api}*:
++
+Moved to `jakarta` namespace. Moved to `jakarta` namespace. Full information in link:https://jakarta.ee/specifications/annotations/2.0/annotations-spec-2.0.html[Jakarta Annotations {version-lib-jakarta-annotations-api} Specification]
+
+* *Jakarta Persistence API {version-lib-jakarta-persistence-api}*:
++
+Moved to `jakarta` namespace. Changes described in link:https://jakarta.ee/specifications/persistence/3.0/jakarta-persistence-spec-3.0.html#revision-history[Jakarta Persistence API {version-lib-jakarta-persistence-api} Specification]
+
+* *Jakarta Transactions API {version-lib-jakarta-transaction-api}*:
++
+Moved to `jakarta` namespace. Changes described in link:https://jakarta.ee/specifications/transactions/2.0/jakarta-transactions-spec-2.0.html#revision-history[Jakarta Transactions API {version-lib-jakarta-transaction-api} Specification]
+
+* *Jakarta WebSocket API {version-lib-jakarta-websockets-api}*:
++
+Moved to `jakarta` namespace. Changes described in link:https://jakarta.ee/specifications/websocket/2.0/websocket-spec-2.0.html#changes[Jakarta WebSocket API {version-lib-jakarta-websockets-api} Specification]
+
+* *Jakarta Bean Validation {version-lib-jakarta-bean-validation}*:
++
+Moved to `jakarta` namespace. Changes described in link:https://jakarta.ee/specifications/bean-validation/2.0/bean-validation_2.0.html#changelog[Jakarta Bean Validation {version-lib-jakarta-bean-validation} Specification]
+
+NOTE: Please, read each specification carefully for incompatible changes!
+
+== Updated CLI
+In version 3.x Helidon CLI brings all benefits of the Project Starter to the command line allowing users to enjoy customizable project generation without leaving the terminal. See it in action xref:{rootdir}/about/cli.adoc[here]!
+
+== Project Starter
+Project Starter is a new web UI for creating Helidon projects. It’s highly customizable and provides various options allowing users to select Helidon features they want to add to the project. At the end users will be able to generate projects up to their taste. Check it out at link:https://helidon.io/starter[Helidon Starter Page]!
+
+== TestNG test support
+
+In Helidon 3.x you have the option to use link:https://testng.org/doc/[TestNG] to test your application. For more information follow the official documents xref:{rootdir}/mp/testing-ng.adoc
+
+== Deprecations
+
+- The custom Helidon OCI clients have been deprecated (link:https://github.com/oracle/helidon/pull/4015[See PR]). Use the OCI Java SDK instead. If you use Helidon MP you can inject OCI SDK clients by adding the dependency `io.helidon.integrations.oci.sdk:helidon-integrations-oci-sdk-cdi`.
```

Comment:
Remove a link to PR. It's too low level for users.

Reply:
removed